lib/Parse/ParseDecl.cpp

```diff
@@ -4699,6 +4699,42 @@ static void diagnoseAndIgnoreObservers(Parser &P,
   }
 }
 
+static StorageImplInfo classifyWithHasStorageAttr(
```

Comment:
@rjmccall This patch introduces a `@_hasStorage` attribute that allows us to break an ambiguity in parseable interfaces.

The mapping we've settled on is an understanding that
```swift
public private(set) var x: Int
```
maps to
```swift
@_hasStorage public var x: Int { get }
```
and
```swift
public var x: Int {
  willSet/didSet {}
}
```
maps to
```swift
@_hasStorage public var x: Int { get set }
```

With that in mind, is this mapping correct?

Reply:
You should be able to rename the `@_sil_stored` attribute for this purpose.

If you're changing this, could you change it to list/expect all the public accessors, i.e. including `modify` and (eventually) `_read`?  You might need to (1) move the logic for deciding the opaque accessors into Sema and then (2) change `StorageImplInfo` to store that bitset.

Reply:
I'm not sure how to handle `_modify` and `_read` if `@_hasStorage` is provided.

What are the semantics of `@sil_stored var x: T { get set }` vs. just `@sil_stored var x: T`? Same for `@sil_stored let x: T { get }` vs. just `@sil_stored let x: T`?

Reply:
Currently it declares that the accessors are actually available.  I'm not sure that's important to know in SIL.  I do think there's a strong argument for it being useful to unambiguously list out the available opaque accessors, but it should be the full set, not just `{get set}` and then inferring `modify` and `read` and so on.  However, to express that properly, you might have to change the representation a bit.

Reply:
I'm going to do this in a future patch, because testing parseable interfaces is currently blocked on this representational change. For now, there's a loud FIXME explaining how it should eventually work.

Reply:
Sounds good.